### PR TITLE
Allow to manipulate buckets on minio

### PR DESCRIPTION
**Pull request by mdellweg**
##### SUMMARY
This allows [{}] to be recognized as an empty tags_list in boto3_tag_list_to_ansible_dict.

##### ISSUE TYPE
- Bugfix Pull Request
- Feature Pull Request

##### COMPONENT NAME
ec2 module_utils

### CODE REVIEW

File: plugins/module_utils/ec2.py [hunk 1]
```diff
@@ -455,7 +455,8 @@ def boto3_tag_list_to_ansible_dict(tags_list, tag_name_key_name=None, tag_value_
     else:
         tag_candidates = {'key': 'value', 'Key': 'Value'}
 
-    if not tags_list:
+    # minio seems to return [{}] as an empty tags_list
+    if not tags_list or not tags_list[0]:
```
Comment (s-hertel) [inline plugins/module_utils/ec2.py:459]: I'd make sure it's entirely empty, since otherwise there should be an error below
```suggestion
    if not tags_list or not any(tag for tag in tags_list):
```

Comment (mdellweg) [reply plugins/module_utils/ec2.py:459]: At this point we could just 'repair' taglist:
```suggestion
    tags_list = [tag for tag in tags_list if tag]
    if not tags_list:
```
This should even provide any further code to choke on '{}'.

Comment (s-hertel) [reply plugins/module_utils/ec2.py:459]: I was thinking the error was helpful for developers so they could see they were passing in a bad list.

Comment (mdellweg) [reply plugins/module_utils/ec2.py:459]: My thinking was being permissive helps to build a more resilient tool (There is not much to misinterpret about an empty dict here, right?).
But i'm fine with either solution. Which one should i pick?

Comment (mdellweg) [reply plugins/module_utils/ec2.py:459]: @jillr @tremble Which one do you prefer?

Comment (tremble) [reply plugins/module_utils/ec2.py:459]: I'm good with either option, However, I would defer to @s-hertel's experience.

We recently added some minimal [unit tests for boto3_tag_list_to_ansible_dict](https://github.com/ansible-collections/amazon.aws/blob/main/tests/unit/module_utils/test_ec2.py#L393).  To reduce the chance of this being reverted by someone unfamiliar with minio would you be willing to add some tests for the 'empty' behaviour?

Should be something like:
```
    def test_boto3_tag_list_to_ansible_dict_empty(self):
        # AWS returns [] when there are no tags
        self.assertEqual(boto3_tag_list_to_ansible_dict([], {})
        # Minio returns [{}] when there are no tags
        self.assertEqual(boto3_tag_list_to_ansible_dict([{}], {})
```

Comment (mdellweg) [reply plugins/module_utils/ec2.py]: @tremble thanks for designing the test for me.

**Comment by tremble** [comment]
Looks like I missed some closing brackets in my example.

File: plugins/module_utils/ec2.py (hunk 1 above)
Comment (tremble) [reply plugins/module_utils/ec2.py]: As someone who's only been contributing to Ansible for a year or so I'm all too aware of how daunting it can be to be asked to write the initial tests for a module or some of the shared code when all you wanted to do was add a minor feature/fix.  I've been trying to improve the situation but it's slow going (especially when writing tests uncovers bugs)

**Comment by mdellweg** [comment]
right fixing it...

File: tests/unit/module_utils/test_ec2.py [hunk 2]
```diff
@@ -394,6 +394,12 @@ def test_boto3_tag_list_to_ansible_dict(self):
         converted_dict = boto3_tag_list_to_ansible_dict(self.tag_example_boto3_list)
         self.assertEqual(converted_dict, self.tag_example_dict)
 
+    def test_boto3_tag_list_to_ansible_dict_empty(self):
+        # AWS returns [] when there are no tags
+        self.assertEqual(boto3_tag_list_to_ansible_dict([], {}))
```
Comment (tremble) [inline tests/unit/module_utils/test_ec2.py:399]:
```suggestion
        self.assertEqual(boto3_tag_list_to_ansible_dict([]), {})
```

File: tests/unit/module_utils/test_ec2.py [hunk 3]
```diff
@@ -394,6 +394,12 @@ def test_boto3_tag_list_to_ansible_dict(self):
...
+        # Minio returns [{}] when there are no tags
+        self.assertEqual(boto3_tag_list_to_ansible_dict([{}], {}))
```
Comment (tremble) [inline tests/unit/module_utils/test_ec2.py:401]:
```suggestion
        self.assertEqual(boto3_tag_list_to_ansible_dict([{}]), {})
```

Comment (mdellweg) [reply tests/unit/module_utils/test_ec2.py]: Sorry, i must have been blind.

Comment (tremble) [reply tests/unit/module_utils/test_ec2.py]: easy mistake to make, nothing to worry about.

**Comment by tremble** [comment]
We have unit tests and there are various integration tests that automatically ran.  I've triggered a new test run, if that passes I'll merge this PR.
